third_party/mbedtls/CMakeLists.txt

```
@@ -26,73 +26,57 @@
 #  POSSIBILITY OF SUCH DAMAGE.
 #
 
-add_library(mbedcrypto)
-
-set(OT_MBEDTLS_DEFAULT_CONFIG_FILE "\"mbedtls-config.h\"")
+set(OT_MBEDTLS_DEFAULT_CONFIG_FILE \"${CMAKE_CURRENT_SOURCE_DIR}/mbedtls-config.h\")
 
 set(OT_MBEDTLS_CONFIG_FILE "" CACHE STRING "The mbedTLS config file")
 
-target_compile_definitions(mbedcrypto
+set(ENABLE_TESTING OFF CACHE BOOL "Disable mbedtls test")
```

Not sure if we should add `FORCE`, because I remember sometimes this doesn't take effect without `FORCE`.
```suggestion
set(ENABLE_TESTING OFF CACHE BOOL "Disable mbedtls test" FORCE)
```

---

@wgtdkp, thoughts?

---

`ENABLE_TESTING` is a global CMake VAR. If it is set by other modules before, then the value will not be changed  here. I think if we want to force disable mbedtls tests, we need 
 `FORCE` (also need to be fixed in OT Commissioner).

For better ways to excludes unwanted targets in submodules, we can add submodules with
```cmake
add_subdirectories(repo EXCLUDE_FROM_ALL)
```
This will not build any targets in `repo` but only those linked or explicitly specified by `ninja <target-name>`.